batchspawner/batchspawner.py

```diff
@@ -467,20 +480,20 @@ class BatchSpawnerRegexStates(BatchSpawnerBase):
         If this variable is set, the match object will be expanded using this string
         to obtain the notebook IP.
         See Python docs: re.match.expand""").tag(config=True)
+    state_unknown_re = Unicode('^$',
```

Comment:
```suggestion
    state_unknown_re = Unicode('',
```
As below, I guess this should be a false-like value if it's not set.

Reply:
If the regex is an empty string, it will match any string and `state_isunknown` will always return `True`. The regex currently implemented will only match empty string, which I think is a good case to conclude the state of the job is unknown when querying the resource manager.

Reply:
Ah, I see what I was missing.  If it was an empty string, in my mind it should be considered unset, and then not checked and it is never "unknown".  I think "unknown"  should be an optional state, and if it's not set, it will simply never return true.